src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/metadata/jdbc/JdbcHelper.java

```diff
@@ -0,0 +1,558 @@
+    }
+
+    public SortedMap<JdbcForeignKeyConstraintMetadata, Collection<JdbcForeignKeyColumnMetadata>>
+            getInversedForeignKeysByTable(DatabaseMetaData metaData, JdbcTableMetadata table)
+                    throws Exception {
+        ResultSet foreignKeys =
+                (table != null)
+                        ? metaData.getExportedKeys(
+                                table.getCatalog(), table.getSchema(), table.getName())
+                        : null;
+        if (foreignKeys != null && foreignKeys.next()) {
+            SortedSetMultimap<JdbcForeignKeyConstraintMetadata, JdbcForeignKeyColumnMetadata>
+                    inversedFkMultimap = TreeMultimap.create();
+            do {
+                JdbcTableMetadata pkTable =
+                        new JdbcTableMetadata(
+                                metaData.getConnection(),
+                                foreignKeys.getString("PKTABLE_CAT"),
+                                foreignKeys.getString("PKTABLE_SCHEM"),
+                                foreignKeys.getString("PKTABLE_NAME"));
+                String pkConstraintName = foreignKeys.getString("PK_NAME");
+                JdbcTableMetadata fkTable =
+                        new JdbcTableMetadata(
+                                metaData.getConnection(),
+                                foreignKeys.getString("FKTABLE_CAT"),
+                                foreignKeys.getString("FKTABLE_SCHEM"),
+                                foreignKeys.getString("FKTABLE_NAME"));
+                JdbcForeignKeyConstraintMetadata pkConstraint =
+                        new JdbcForeignKeyConstraintMetadata(pkTable, pkConstraintName, fkTable);
+
+                // get FKColumn from table just in order to get datatype
+                AttributeMetadata aColumn =
+                        this.getColumnFromTable(
+                                metaData, fkTable, foreignKeys.getString("FKCOLUMN_NAME"));
+                String columnType = aColumn.getType();
+
+                JdbcForeignKeyColumnMetadata fkColumns =
+                        new JdbcForeignKeyColumnMetadata(
+                                fkTable,
+                                foreignKeys.getString("FKCOLUMN_NAME"),
+                                columnType,
+                                new JdbcColumnMetadata(
+                                        pkTable,
+                                        foreignKeys.getString("PKCOLUMN_NAME"),
+                                        columnType,
+                                        false));
+                inversedFkMultimap.put(pkConstraint, fkColumns);
+            } while (foreignKeys.next());
+            SortedMap<JdbcForeignKeyConstraintMetadata, Collection<JdbcForeignKeyColumnMetadata>>
+                    inversedFkMap = new TreeMap<>();
+            inversedFkMap.putAll(inversedFkMultimap.asMap());
+            return inversedFkMap;
+        }
+        return null;
```

Comment:
This method should declare on comment that will return null and on which case

Reply:
fixed